functional-samples/ai.gemini-on-device-alt-texter/background.js

```diff
@@ -0,0 +1,49 @@
+chrome.runtime.onInstalled.addListener(() => {
+  chrome.contextMenus.create({
+    id: 'generateAltText',
+    title: 'Generate alt text',
+    contexts: ['image']
+  });
+});
+
+let openPopupPromise;
```

Comment:
Can we declare this in `chrome.contextMenus.onClicked`? I don't think it needs to be a global variable for any reason. Alternatively, if we move the prompt API call into a separate function, we could do something like:

```
const [result] = await Promise.all([
  getResponse(info.srcUrl),
  chrome.action.openPopup()
]);
chrome.runtime.sendMessage(...);
```

That would remove the need for _both_ the `result` variable and the `openPopupPromise` one.